src/core/packages/saved-objects/import-export-server-internal/src/export/saved_objects_exporter.test.ts

```diff
@@ -1039,6 +1039,130 @@ describe('getSortedObjectsForExport()', () => {
         })
       );
     });
+
+    // Unskip after: https://github.com/elastic/kibana/issues/242671
+    describe.skip('access control', () => {
+      test('applies the access control transform if defined', async () => {
+        // const accessControlExportTransform: SavedObjectsExportTransform = (ctx, objects) => {
+        //   objects.forEach((obj: SavedObject<any>) => {
+        //     if (typeRegistry.supportsAccessControl(obj.type))
+        //       obj.attributes.foo = 'modified by access control transform';
+        //   });
+        //   return objects;
+        // };
+
+        typeRegistry.registerType({
+          name: 'foo',
+          mappings: { properties: {} },
+          namespaceType: 'single',
+          hidden: false,
+          supportsAccessControl: true,
+          management: {
+            importableAndExportable: true,
+            onExport: (ctx, objects) => {
+              objects.forEach((obj: SavedObject<any>) => {
+                obj.attributes.foo = 'modified';
+              });
+              return objects;
+            },
+          },
+        });
+        exporter = new SavedObjectsExporter({
+          exportSizeLimit,
+          logger,
+          savedObjectsClient,
+          typeRegistry,
+        });
+
+        savedObjectsClient.find.mockResolvedValueOnce({
+          total: 1,
+          saved_objects: [
+            {
+              id: '1',
+              type: 'foo',
+              attributes: {
+                foo: 'initial',
+              },
+              score: 0,
+              references: [],
+            },
+          ],
+          per_page: 1,
+          page: 0,
+        });
+        const exportStream = await exporter.exportByTypes({
+          request,
+          types: ['foo'],
+          excludeExportDetails: true,
+        });
+
+        const response = await readStreamToCompletion(exportStream);
+
+        expect(response).toHaveLength(1);
+        expect(response[0].attributes.foo).toEqual('modified by access control transform');
+      });
+
+      test('does not apply the access control transform to types that do not support access control', async () => {
+        // const accessControlExportTransform: SavedObjectsExportTransform = (ctx, objects) => {
```

Comment:
uncomment the transform. Without it, we're not actually testing what it claims to.